pkg/core/native/native_neo.go

```diff
@@ -444,8 +501,52 @@ func (n *NEO) SetGASPerBlock(ic *interop.Context, index uint32, gas *big.Int) (b
 	return true, n.putGASRecord(ic.DAO, index, gas)
 }
 
-// CalculateBonus calculates amount of gas generated for holding `value` NEO from start to end block.
-func (n *NEO) CalculateBonus(ic *interop.Context, value *big.Int, start, end uint32) (*big.Int, error) {
+func (n *NEO) checkCandidate(d dao.DAO, pub *keys.PublicKey, c *candidate) error {
+	if c.Registered || c.Votes.Sign() != 0 {
+		return nil
+	}
+	if err := d.DeleteStorageItem(n.ContractID, makeValidatorKey(pub)); err != nil {
+		return err
+	}
+
+	var toRemove []string
+	d.Seek(n.ContractID, makeVoterKey(pub.Bytes()), func(k, v []byte) {
+		toRemove = append(toRemove, string(k))
+	})
+	for i := range toRemove {
+		if err := d.DeleteStorageItem(n.ContractID, []byte(toRemove[i])); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func makeVoterKey(pub []byte) []byte {
+	key := make([]byte, 38)
```

Comment:
```suggestion
	key := make([]byte, 34, 38)
```
I don't think current code is correct.

Reply:
Fixed.